pkg/uhttp/wrapper.go

```diff
@@ -180,8 +180,18 @@ func NewBaseHttpClientWithContext(ctx context.Context, httpClient *http.Client,
 // status code 204 No Content), then pass a `nil` to `response`.
 func WithJSONResponse(response interface{}) DoOption {
 	return func(resp *WrapperResponse) error {
-		if !IsJSONContentType(resp.Header.Get(ContentType)) {
-			return fmt.Errorf("unexpected content type for json response: %s", resp.Header.Get(ContentType))
+		contentHeader := resp.Header.Get(ContentType)
+		if contentHeader == "" {
```

Comment:
An empty string will cause IsJSONContentType() to return false, so this if statement isn't needed.

Reply:
but what if it is something valid like `text/plain`?

Reply:
IsJSONContentType() only returns true if content type starts with "application" and contains "json". If it's "text/plain", then we'll error and log the body.

Though I do notice that IsJSONContentType() should probably normalize the content type string like IsXMLContentType() does.

Reply:
we prefer the value to remain empty then, got it.